File: EasyMSXPython/Order.py
# Order.py

from Fields import Fields

class Order:
    
    def __init__(self, parent):
        self.parent = parent
        self.sequence = 0
        self.notificationHandlers = []
        self.fields=Fields(self)
        
    def field(self,fieldname):
        return self.fields.field(fieldname)

    def addNotificationHandler(self,handler):
        self.notificationHandlers.append(handler)

    def notify(self, notification):
        for h in self.notificationHandlers:
            if not notification.consumed:
                h(notification)
        if not notification.consumed: 
            self.parent.notify(notification)
            

File: CSharp/cs_EasyMSX-master/EasyMSX/Order.cs
using System.Collections.Generic;

namespace com.bloomberg.emsx.samples {

    public class Order : FieldsOwner {
	
	    internal Orders parent;
	    internal int sequence;
	    private Broker broker;
	
	    List<NotificationHandler> notificationHandlers = new List<NotificationHandler>();

	    internal Order(Orders parent) {
		    this.parent = parent;
		    this.fields = new Fields(this);
		    this.sequence = 0;
	    }
	
	    public Broker getBroker(){
		    return this.broker;
	    }
	
	    public Field field(string fieldname) {
		    return this.fields.field(fieldname);
	    }

	    public void addNotificationHandler(NotificationHandler notificationHandler) {
		    notificationHandlers.Add(notificationHandler);
	    }

	    internal void notify(Notification notification) {
		
		    foreach(NotificationHandler nh in notificationHandlers) {
			    if(!notification.consume) nh.processNotification(notification);
		    }
		    if(!notification.consume) parent.processNotification(notification);

	    }
    }
}